Editor/Scripts/interactivetutorials_dialog.py

```diff
@@ -46,6 +43,36 @@ def paintEvent(self, event):
         pen.setWidth(self.border_width)
         painter.setPen(pen)
         painter.drawRect(event.rect())
+        #print(self.width)
+        #print(self.height)
+
+        #option 1: highlight the content of the widget
+        if self.width < 50 and self.height < 50:
+            painter.setOpacity(0.3)
+            painter.fillRect(event.rect(), QBrush(QColor ("blue")))
+
+        #option 2: highlight the area around the widget not including the widget
+        if self.width < 50 and self.height < 50:
+            painter.setOpacity(0.3)
+            topLeftX = event.rect().topLeft().x
+            #topRightX = event.rect().topRight().x
+            bottomLeftX = event.rect().bottomLeft().x
+            bottomRightX = event.rect().bottomRight().x
+            topLeftY = event.rect().topLeft().y
+            #topRightY = event.rect().topRight().y
+            bottomLeftY = event.rect().bottomLeft().y
+            #bottomRightY = event.rect().bottomRight().y
+            #draw 4 rectangles around "self"
+            painter.fillRect(bottomLeftX - self.width, bottomLeftY, self.width, self.height, QBrush(QColor ("blue")))
+            painter.fillRect(bottomRightX, bottomLeftY, self.width, self.height, QBrush(QColor ("blue")))
+            painter.fillRect(topLeftX - self.width, topLeftY, self.width * 3, self.height, QBrush(QColor ("blue")))
+            painter.fillRect(topLeftX - self.width, bottomLeftY - self.height, self.width * 3, self.height, QBrush(QColor ("blue")))       
+
+        #option 3: add a pulse effect
+        if self.width < 50 and self.height < 50:
+            update_widget(self, event)           
```

Comment:
The `update_widget` only needs to be called when you are choosing a new widget/item to highlight. I think what you meant to use was `update` which will trigger the highlight to be repainted. Instead of calling it in the `paintEvent` method though, you should call it outside of here on a timed loop at a certain interval (maybe 500ms?)